engine/src/main/java/net/jqwik/engine/properties/configurators/UniqueElementsConfigurator.java

```diff
@@ -63,13 +63,28 @@ private <T> List<T> toList(Iterator<T> i) {
 		return list;
 	}
 
-	private boolean isUnique(Collection<?> list, Function<Object, Object> extractor) {
-		Set<Object> set = list.stream().map(extractor).collect(CollectorsSupport.toLinkedHashSet());
-		return set.size() == list.size();
+	private static <C extends Collection<? extends Object>> Predicate<C> isUnique(UniqueElements uniqueness) {
+		Class<? extends Function<?, Object>> extractorClass = uniqueness.by();
+		if (extractorClass.equals(UniqueElements.NOT_SET.class)) {
+			// intentionally use `items.getClass().equals(HashSet.class)`
+			// instead of `items instanceof HashSet`, because subclasses
+			// of HashSet may break Set conventions.
+			return items -> items.getClass().equals(HashSet.class)
+								|| items.size() == items.stream().distinct().count();
```

Comment:
This is a suboptimal way to check uniqueness. You can stop as soon as you detect the first non-unique element.

At the same time, elements in hashset are always unique, aren't they?

Reply:
Oh You're right, it would be much faster to stop early. I will fix that.

Yes that's why added the `HashSet` clause. It might speed things up sometimes.

Reply:
Changed the logic. Now it should stop early.

Reply:
Nice. There's one more case doing `stream.distinct.count`: https://github.com/jqwik-team/jqwik/blob/124d7af3d56fabfb5b5bf182c339cb3bff3825f9/engine/src/main/java/net/jqwik/engine/properties/FeatureExtractor.java#L32
I wonder if it makes sense to unify them.

WDYT?